packages/experiment-tag/src/experiment.ts

```diff
@@ -121,14 +169,34 @@ export const initializeExperiment = (apiKey: string, initialFlags: string) => {
   globalScope.webExperiment.addPlugin(globalScope.experimentIntegration);
   globalScope.webExperiment.setUser(user);
 
-  const variants = globalScope.webExperiment.all();
-
   setUrlChangeListener();
-  applyVariants(variants);
+
+  // apply local variants
+  applyVariants(globalScope.webExperiment.all(), localFlagKeys);
+
+  if (!isRemoteBlocking) {
+    // Remove anti-flicker css if remote flags are not blocking
+    globalScope.document.getElementById?.('amp-exp-css')?.remove();
+  }
+
+  if (remoteFlagKeys.size === 0) {
+    return;
+  }
+
+  try {
+    await globalScope.webExperiment.doFlags();
+    // apply remote variants
+    applyVariants(globalScope.webExperiment.all(), remoteFlagKeys);
+  } catch (error) {
+    console.warn('Error fetching remote flags:', error);
```

Comment:
I think we should still apply local variants in the case that the fetch fails. What's your thought here?

Reply:
Meaning to evaluate the remote flags with local properties? I would avoid just to prevent variant jumping.